### ISSUE TITLE
[indent] parameter with decorator wrongly indented

### ISSUE THREAD

**Issue by edward-soetiono-asto** [typescript-eslint/typescript-eslint#1232]
**Repro**

```JSON
{
  "extends": ["airbnb-typescript/base", "prettier"],
  "parser": "@typescript-eslint/parser",
  "plugins": ["import", "@typescript-eslint", "prettier"],
  "rules": {
    "import/prefer-default-export": "off",
    "max-classes-per-file": "off",
    "class-methods-use-this": "off",
    "import/no-unresolved": "error"
  },
  "settings": {
    "import/parsers": {
      "@typescript-eslint/parser": [".ts"]
    },
    "import/resolver": {
      "typescript": {}
    }
  }
}

```

```TS
@Post('/')
@OnUndefined(201)
async someFunction(
  @CurrentUser({ required: true }) user: User,
  @Body({ required: true }) body: SomeRequest
): Promise<void> {
  // function truncated
}
```

**Expected Result**

```TS
@Post('/')
@OnUndefined(201)
async someFunction(
  @CurrentUser({ required: true }) user: User,
  @Body({ required: true }) body: SomeRequest
): Promise<void> {
  // function truncated
}
```

**Actual Result**

```TS
@Post('/')
@OnUndefined(201)
async someFunction(
  @CurrentUser({ required: true }) user: User,
    @Body({ required: true }) body: SomeRequest
): Promise<void> {
  // function truncated
}

```

**Additional Info**

The 2nd parameter `@Body` was wrongly indented 6 spaces ( should be 4 spaces).

Works in `@typescript-eslint/eslint-plugin` and `@typescript-eslint/parser` 2.7.0

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `2.8.0` |
| `@typescript-eslint/parser`        | `2.8.0` |
| `TypeScript`                       | `3.7.2` |
| `ESLint`                           | `6.6.0` |

**Comment by sgarner** [comment]
Looks like this bug was an unwanted side effect introduced by #1189.

Using the following configuration:

```json
{
  "rules": {
    "@typescript-eslint/indent": ["error", 2]
  }
}
```

✅  Under eslint-plugin@=2.7.0 the following indentation passes linting, as desired:

```ts
export default class CreateMigration extends Command {
  public execute(
    @param({ description: 'Migration name', required: true, type: String })
      name: string,
    @param({ description: 'Timestamp (optional, default will use current time)', required: false, type: Number })
      timestamp?: number,
  ): void {
    return;
  }
}
```

❌  Under eslint-plugin@^2.8.0 the following error is thrown:

```
  5:1  error  Expected indentation of 6 spaces but found 4  @typescript-eslint/indent
```

The indentation it expects looks, nonsensically, like the following:

```ts
export default class CreateMigration extends Command {
  public execute(
    @param({ description: 'Migration name', required: true, type: String })
      name: string,
      @param({ description: 'Timestamp (optional, default will use current time)', required: false, type: Number })
      timestamp?: number,
  ): void {
    return;
  }
}
```

It is not allowed to have the decorators aligned with the parameters either, as in:

```ts
export default class CreateMigration extends Command {
  public execute(
    @param({ description: 'Migration name', required: true, type: String })
    name: string,
    @param({ description: 'Timestamp (optional, default will use current time)', required: false, type: Number })
    timestamp?: number,
  ): void {
    return;
  }
}
```

❌  This produces the following errors:

```
  4:1  error  Expected indentation of 6 spaces but found 4  @typescript-eslint/indent
  5:1  error  Expected indentation of 6 spaces but found 4  @typescript-eslint/indent
  6:1  error  Expected indentation of 6 spaces but found 4  @typescript-eslint/indent
```

**Comment by bradzacher** [comment]
Happy to accept a PR.
There's good reasons that the indent rule is buggy and problematic:
1) None of the maintainers use it, because we all use prettier.
2) Our extension implementation monkey patches the base rule, and does it poorly.
3) We monkey patch the rule, [because the original implementation is 1600 lines of complexity](https://github.com/eslint/eslint/blob/master/lib/rules/indent.js).

I looked into converting the rule to a first-class implementation (hence [this unused code exists in the package](https://github.com/typescript-eslint/typescript-eslint/tree/master/packages/eslint-plugin/src/rules/indent-new-do-not-use)), but as per reasons (1) and (3), it's more work than I have time to do, when I have no need for the rule.

I've always offered the same response:
This is a community run project, and we rely upon community contributions. The core maintainers are volunteers, and use their time mostly on moderating and reviewing.
If you (or someone else) cares about this rule enough, happy to accept a PR to either fix the monkey patch, or to make the rule a first-class citizen.

**Comment by colthreepv** [comment]
I agree original implementation it's a bit crazy, I think all the turmoil regarding this rule is that, everyone tries not to disable it, because **it is responsible to indent the code of your entire codebase**

Maybe it would be just ok to start from scratch supporting very few use cases for indent, scrapping the original `eslint` one?

**Comment by bradzacher** [comment]
Merging into #1824
